lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -543,31 +532,15 @@ void EscapeAnalysis::ConnectionGraph::initializePointsTo(CGNode *initialNode,
       });
     }
     // For all nodes visited in step 1, if any node was not backward-reachable
-    // from a pointsTo edge, create an edge for it and restart traversal.
-    //
-    // First process all forward-reachable nodes in backward order, then process
-    // all backwardReachable nodes in forward order.
-    while (nextUpdatedNodeIdx != updatedNodes.size()) {
-      CGNode *node = updatedNodes[nextUpdatedNodeIdx];
-      // When processBackwardReachable == true, the backwardReachable set is
-      // empty and all forward reachable nodes already have a pointsTo edge.
-      if (!backwardReachable.count(node)) {
-        if (!node->pointsTo) {
-          pointsToEdges.push_back(node);
-          break;
-        }
-      }
-      if (processBackwardReachable) {
-        ++nextUpdatedNodeIdx;
-        continue;
-      }
-      if (nextUpdatedNodeIdx > 0) {
-        --nextUpdatedNodeIdx;
-        continue;
+    // from a pointsTo edge, create an edge for it and restart traversal. This
+    // only happens when step 1 fails to find leaves in the defer web because
+    // of defer edge cycles.
+    while (!updatedNodes.empty()) {
+      CGNode *node = updatedNodes.nodeVector.pop_back_val();
+      if (!node->pointsTo) {
+        pointsToEdges.push_back(node);
+        break;
       }
-      // reverse direction
-      backwardReachable.clear();
-      processBackwardReachable = true;
     }
     // This outer loop is exceedingly unlikely to execute more than twice.
```

Comment:
Actually, the outer loop can only run twice. After the first iteration, updateNodes is empty. Which prevents adding nodes to "pointsToEdges" in the second iteration.

Reply:
I fixed this comment
```
    // For all nodes visited in step 1, pick a single node that was not
    // backward-reachable from a pointsTo edge, create an edge for it and
    // restart traversal. This only happens when step 1 fails to find leaves in
    // the defer web because of defer edge cycles.
```
The outer loop can repeat for every pointsTo edge that needs to be created and wasn't a "leaf" node or part of another cycle. So, this only happens when there are multiple independent cycles in the same web. Improbable but possible.